File: oop/project0/Point.java
package com.java.project0;

public class Point {
    private char name;
    private double x;
    private double y;

    public Point() {
        this.name = 'O';
        this.x = 0;
        this.y = 0;
    }

    public Point(char name, double x, double y) {
        this.name = name;
        this.x = x;
        this.y = y;
    }

    public void display() {
        System.out.println(name + "(" + x + ", " + y + ")");
    }

    public void translate(double dx, double dy) {
        this.x += dx;
        this.y += dy;
    }

    public Point midpoint(Point other) {
        double mx = (this.x + other.x) / 2;
        double my = (this.y + other.y) / 2;
        return new Point('M', mx, my);
    }

    public Point symmetric() {
        return new Point('S', -this.x, -this.y);
    }

    public double distance(Point other) {
        return Math.sqrt(Math.pow(other.x - this.x, 2) + Math.pow(other.y - this.y, 2));
    }

    public boolean compare(Point other) {
        return (this.x == other.x && this.y == other.y);
    }
}

File: oop/project1/Point.cpp
#include "Point.h"

#include <cmath>
#include <iostream>

Point::Point() : name('O'), x(0), y(0) {}

Point::Point(char name, double x, double y) : name(name), x(x), y(y) {}

void Point::display() const {
    std::cout << name << "(" << x << ", " << y << ")" << std::endl;
}

void Point::translate(double dx, double dy) {
    x += dx;
    y += dy;
}

Point Point::midpoint(const Point& other) const {
    double mx = (x + other.x) / 2;
    double my = (y + other.y) / 2;
    return Point('M', mx, my);
}

Point Point::symmetric() const {
    return Point('S', -x, -y);
}

double Point::distance(const Point& other) const {
    return std::sqrt(std::pow(other.x - x, 2) + std::pow(other.y - y, 2));
}

bool Point::compare(const Point& other) const {
    return x == other.x && y == other.y;
}
